### PR TITLE
Feature/fix beneficiary creation

### PR DESCRIPTION
**Pull request by AntoineAresu**
[Ticket 1](https://trello.com/c/p0baVDbj/3506-cr%C3%A9ation-new-b%C3%A9n%C3%A9ficiaire)[
Ticket 2](https://trello.com/c/dwjJsNu7/3505-affiliation-b%C3%A9n%C3%A9ficiaire-existant)

### CODE REVIEW

File: assets/controllers/password_generator_controller.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export default class extends Controller {
+  static targets = ['input'];
+
+  generate() {
+    this.inputTarget.value = this.getGeneratedPassword(8);
+  }
+
+  getGeneratedPassword(length) {
+    let result = '';
+    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
```
Comment (tcheymol) [inline assets/controllers/password_generator_controller.js:13]: Should be class const

File: assets/controllers/password_generator_controller.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    const charactersLength = characters.length;
```
Comment (tcheymol) [inline assets/controllers/password_generator_controller.js:14]: No need for this variable

File: assets/controllers/password_generator_controller.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    let i = 0;
+
+    while (i < length) {
+      result += characters.charAt(Math.floor(Math.random() * charactersLength));
+      i++;
+    }
+
+    return result;
```
Comment (tcheymol) [inline assets/controllers/password_generator_controller.js:22]: you could be doing something more functionnal style like

```es6
return (new Array(characters.length))
  .reduce((acc) => acc + characters.charAt(Math.floor(Math.random() * characters.length)), '');
```

or even mofe a bit of code in a function
```
const getRandomByte() => this.characters.length.charAt(Math.floor(Math.random() * this.characters.length));

...

return (new Array(characters.length))
  .reduce((acc) => acc + this->getRandomByte(), '');
```

File: src/FormV2/UserCreation/CreateBeneficiaryType.php [hunk 4]
```diff
@@ -67,12 +68,25 @@ public function addStep1Fields(FormBuilderInterface $builder, ?Beneficiaire $ben
...
     {
         $builder
             ->add('password', TextType::class, [
                 'property_path' => 'user.plainPassword',
                 'label' => 'password',
+                'attr' => [
+                    'data-password-generator-target' => 'input',
+                ],
+            ])
+            ->add('generatePassword', ButtonType::class, [
```
Comment (tcheymol) [inline src/FormV2/UserCreation/CreateBeneficiaryType.php:81]: I am not sure about using these in the form type

**Review by tcheymol — APPROVED** [review]

**Review by NilsRec — APPROVED** [review]

**Pull request merged**